front/lib/swr/vaults.ts

```diff
@@ -131,3 +131,33 @@ export function useVaultDataSourceViews<
     isVaultDataSourceViewsError: error,
   };
 }
+
+export function useSystemVault({
+  workspaceId,
+  disabled = false,
+}: {
+  workspaceId: string;
+  disabled?: boolean;
+}) {
+  const systemVaultFetcher: Fetcher<GetVaultsResponseBody> = fetcher;
+
+  const { data, error, mutate } = useSWRWithDefaults(
+    `/api/w/${workspaceId}/vaults?role=admin`,
+    systemVaultFetcher,
+    { disabled }
+  );
+
+  const systemVault = useMemo(() => {
+    if (!data) {
+      return null;
+    }
+    return data.vaults.find((v) => v.kind === "system") || null;
```

Comment:
Indeed not super elegant to load them all. We don't necessarily need to overcomplicate and keep a GetVaultsResponseBody but expect a result of 1 with only the system vault on this specific hook  and ping `/api/w/${workspaceId}/vaults?kind=system`.

Since we only need it here for now I don't think it's worth having it accessible from auth like we have auth.group().